src/amici.cpp

```diff
@@ -1383,6 +1384,375 @@ int getDiagnosisB(int it, void *ami_mem, UserData *udata, ReturnData *rdata, Tem
+    N_VDestroy_Serial(ns_s);
+    N_VDestroy_Serial(ns_r);
+    N_VDestroy_Serial(ns_rr);
+    N_VDestroy_Serial(ns_rt);
+    N_VDestroy_Serial(ns_v);
+    N_VDestroy_Serial(ns_Jv);
+    N_VDestroy_Serial(ns_tmp);
+    N_VDestroy_Serial(ns_Jdiag);
+    
+    N_VScale(-1.0, tdata->xdot, tdata->xdot);
+    
+    // Return
+    return(stat1);
+ }
+
+/* ------------------------------------------------------------------------------------- */
+/* ------------------------------------------------------------------------------------- */
+/* ------------------------------------------------------------------------------------- */
+
+int getNewtonOutput(UserData *udata, TempData *tdata, ReturnData *rdata, int newton, double run_time) {
+    /**
+     * getNewtonOutput stores the output of the Newton solver run.
+     *
+     * @param[in] udata pointer to the user data struct @type UserData
+     * @param[in] tdata pointer to the temporary data struct @type TempData
+     * @param[out] rdata pointer to the return data struct @type ReturnData
+     * @param[in] newton integer flag indicating the run of the Newton solver
+     * @param[in] clock_t starttime timer running from the start of Newton solver
+     * @return int status flag indicating success of execution @type int
+     */
+    
+    realtype *x_tmp;
+    
+    // Get time for Newton solve
+    rdata->newton_time[0] = run_time;
+    
+    // Since the steady state was found, current time is set to infinity
+    tdata->t = INFINITY;
+    
+    // Write output
+    x_tmp = N_VGetArrayPointer(tdata->x);
+    for (int ix=0; ix<udata->nx; ix++) {
+        rdata->xss[ix] = x_tmp[ix];
+    }
+    
+    // Write flag for the Newton solver
+    *rdata->newton = (double) newton;
+    
+    return(AMICI_SUCCESS);
+}
+
+/* ------------------------------------------------------------------------------------- */
+/* ------------------------------------------------------------------------------------- */
+/* ------------------------------------------------------------------------------------- */
+
+int getNewtonSimulation(void *ami_mem, UserData *udata, TempData *tdata, ReturnData *rdata) {
+    /**
+     * getNewtonSimulation solves the forward problem, if the first Newton solver run did not succeed.
+     *
+     * @param[in] udata pointer to the user data struct @type UserData
+     * @param[in] tdata pointer to the temporary data struct @type TempData
+     * @param[out] rdata pointer to the return data struct @type ReturnData
+     * @param[in] newton integer flag indicating the run of the Newton solver
+     * @param[in] clock_t starttime timer running from the start of Newton solver
+     * @return int status flag indicating success of execution @type int
+     */
+    
+    
+    double res_abs;
+    double res_rel;
+    double sim_time;
+    int status = (int) *rdata->status;
+    realtype *x_tmp;
+    N_Vector rel_x = N_VNew_Serial(udata->nx);
+    N_Vector tmp_x = N_VNew_Serial(udata->nx);
```

Comment:
To have both `tmp_x` and `x_tmp` is somewhat confusing and seems like to attract bugs.